src/checks/checks-common.ts

```diff
@@ -0,0 +1,56 @@
+/* eslint-disable @typescript-eslint/camelcase */
+export interface GitData {
+  org: string
+  repo: string
+  sha: string
+}
+
+export interface CheckResult {
+  conclusion: Level
+  output: CheckOutput
+}
+
+export interface CheckOutput {
+  title: string
+  summary: string
+  annotations?: Annotation[]
+  text?: string
+}
+
+export interface Annotation {
+  path: string
+  blob_href?: string
+  start_line?: number
+  end_line?: number
+  annotation_level: Level
+  message: string
+  raw_details: string
+  title?: string
+}
+
+export type Level = 'success' | 'failure' | 'neutral' | 'notice' | 'warning'
+
+export const printSummary = (checkResult: CheckResult, ci?: boolean): void => {
```

Comment:
we normally using the function style for the backend projects:

export function printSummary(checkResult: CheckResult, ci?: boolean): void {
...
}